### ISSUE TITLE
Add Advanced Post Cache plugin

### ISSUE THREAD

**Issue by roborourke** [humanmade/altis-cms#207]
Add https://github.com/Automattic/advanced-post-cache

This is blocked by a8c not having published the package to packagist yet.

Acceptance criteria:

- [ ] Advanced post cache is required in `composer,json`
- [ ] Advanced post cache file is loaded on the `muplugins_loaded` hook in `inc/namespace.php`
- [ ] The Advanced post cache package is referenced in the `install-overrides` array in `composer.json`

**Comment by PriyaBhatia0210** [comment]
Blocked by upstream

**Comment by rmccue** [comment]
Looks like this is unblocked: https://packagist.org/packages/automattic/advanced-post-cache

**Comment by robindevitt** [comment]
> Above plugin are added to the exclude list in humanmade/altis-core here https://github.com/humanmade/altis-core/blob/master/inc/composer/class-installer.php#L26

Should this be the correct URL : https://github.com/humanmade/altis-core/blob/master/inc/composer/class-override-installer.php ?

**Comment by roborourke** [comment]
@robindevitt that's now a legacy list, you need to add the package name to the list in the `installer-overrides` section of this repo's `composer.json` instead. I'll get those docs updated.

**Comment by robindevitt** [comment]
@roborourke From your [comment in slack](https://hmn.slack.com/archives/C5WSNGC9F/p1636116975164900?thread_ts=1636106496.144100&cid=C5WSNGC9F) around these lines https://github.com/humanmade/altis-cms/blob/master/composer.json#L43-L48

1. Are those strings relevant to the GitHub URL's or to a folder for each item?
2. I have added a string for `"Automattic/advanced-post-cache"` inside of `install-overrides`, is there anything I need to "run" to get it to install so I will see the field in the vendor folder?

**Comment by roborourke** [comment]
@robindevitt 

> Are those strings relevant to the GitHub URL's or to a folder for each item?

They represent the composer package name which is set by the `name` property in a given repo's `composer.json` file, in this case it is `automattic/advanced-post-cache`: https://github.com/Automattic/advanced-post-cache/blob/master/composer.json#L2

Note that it is lower case.

> I have added a string for "Automattic/advanced-post-cache" inside of install-overrides, is there anything I need to "run" to get it to install so I will see the plugin in the vendor folder?

The package also needs to be added to the module's `composer.json` in the `require` section to actually add the dependency. It appears that they haven't actually created any tags on that repo so we might need to set the version requirement to the latest commit -you'll need to look at the composer docs for how to do that.

If you can add an issue on their repo to create a tag it'd be good.

**Comment by robindevitt** [comment]
@roborourke 

>  Advanced post cache is required in composer,json
>  Advanced post cache file is loaded on the muplugins_loaded hook in inc/namespace.php

 I have committed these two changes and created a new PR. I had the versioning for it earlier but didn't have it set up as `dev-master`. I have used `master` on it's own.

> Above plugin are added to the exclude list in humanmade/altis-core here https://github.com/humanmade/altis-core/blob/master/inc/composer/class-installer.php#L26

This item I am a little bit confused about...
1. I suspect the correction location for this is: https://github.com/humanmade/altis-core/blob/master/inc/composer/class-override-installer.php
2. I'm then going to add it to the the following array : https://github.com/humanmade/altis-core/blob/f8013c722ede5aab57938e84c9fa3440f0dc2869/inc/composer/class-override-installer.php#L61-L95
3. This will need to be created in a new issue with in the `altis-core` repo. Do I then reference this issue on the PR to link the two?

**Comment by roborourke** [comment]
Ace, yeah with composer when referencing a branch you must always prefix it with `dev-`.

You don't need to do the part referencing `class-override-installer.php`, I explained it badly but it's the legacy approach to changing the installation directory and no longer relevant. I need to updates those internal docs.

**Comment by robindevitt** [comment]
Hey @roborourke 

> If you can add an issue on their repo to create a tag it'd be good.

Is this a tag for the version number that would be in the `composer.json` file? I tried to reference some of the other repos to get an idea of what the "tag" should be but came to a dead-end.

**Comment by roborourke** [comment]
This is blocked again so I'll move it back to the backlog for now.

**Comment by rmccue** [comment]
WordPress now includes a query cache in core, so we do not need to implement this.
